test/fail_compilation/retscope.d

```diff
@@ -267,3 +267,34 @@ void escape5() @safe
     pp = &p; // error
 }
 
+/***********************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope.d(287): Error: escaping reference to local variable b
+---
+*/
+
+@safe int* foo6()(int* arg)
+{
+    return arg;
+}
+
+int* escape6() @safe
+{
+    int b;
+    return foo6(&b);
+}
+
+/***************************************************/
+
+struct S7
+{
+    int[10] a;
+    int[3] abc(int i)
```

Comment:
This function is not `@safe`, hence `scope` is not enforced here. I got bit by this many times, hence why I so strongly argued for `scope` being enforced everywhere.

I would also add the following test cases:
```D
int[3] test1 (scope int[] p) @safe { return p[0 .. 3]; }             // Do not error because there isn't any indirection
char*[3] test1 (scope char*[] p) @safe { return p[0 .. 3]; }   // Error here because there is an indirection
```